test/rethinkdb/joins_test.clj

```diff
@@ -0,0 +1,51 @@
+(ns rethinkdb.joins-test
+  (:require [clojure.test :refer :all]
+            [rethinkdb.query :as r]
+            [rethinkdb.test-utils :as u]))
+
+(def names [{:id 1 :name "Pikachu", :type 2},
+            {:id 2 :name "Magnemite", :type [1 2]}])
```

Comment:
That's also unclear to me. I thought that the eq-join would map the types in the array, but it doesn't. I can look into this in more details, just to confirm.